### PR TITLE
Snapshots: Change default expiration

### PR DESCRIPTION
**Pull request by evictorero**
**What is this feature?**
Improving awareness about the Snapshots being public changing CTA text. Change default expiration ensuring snapshots are only available for a fixed period unless the user explicitly decides to change it to `never`.

* Changed CTA text from `Local Snapshot` to `Publish Snapshot`
* Changed `7 Days` to `1 Week`
* Changed default from `Never` to `1 Week`

Fixes https://github.com/grafana/grafana-enterprise/issues/6209

**Special notes for your reviewer:**
Changes were made both in the old sharing modal and the new scenes sharing modal.

Scenes  
Screenshot 2024-02-27 at 18 20 48

Old Dashboards
Screenshot 2024-02-27 at 18 10 01

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 1]
```diff
@@ -49,15 +45,19 @@ export class ShareSnapshot extends PureComponent<Props, State> {
         value: 60 * 60 * 24,
       },
       {
-        label: t('share-modal.snapshot.expire-week', `7 Days`),
+        label: t('share-modal.snapshot.expire-week', `1 Week`),
```
Comment (juanicabanas) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:48]:
```suggestion
        label: t('share-modal.snapshot.expire-week', '1 Week'),
```

File: public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx [hunk 2]
```diff
@@ -49,15 +45,19 @@ export class ShareSnapshot extends PureComponent<Props, State> {
...
         value: 60 * 60 * 24 * 7,
       },
+      {
+        label: t('share-modal.snapshot.expire-never', `Never`),
```
Comment (juanicabanas) [inline public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx:52]:
```suggestion
        label: t('share-modal.snapshot.expire-never', 'Never'),
```

**Review by juanicabanas — APPROVED** [review]
LGTM!
What about the already translated texts that were modified?
Will it be modified by the corresponding team?

**Comment by evictorero** [comment]
> LGTM! What about the already translated texts that were modified? Will it be modified by the corresponding team?

Yes, I asked about it [here](https://raintank-corp.slack.com/archives/C025WTVRBSN/p1709038723748209) if you are interested

**Pull request merged**